src/main/java/com/github/manolo8/darkbot/gui/tree/editors/ImagePicker.java

```diff
@@ -0,0 +1,76 @@
+package com.github.manolo8.darkbot.gui.tree.editors;
+
+import com.github.manolo8.darkbot.config.ImageWrapper;
+import com.github.manolo8.darkbot.gui.AdvancedConfig;
+import eu.darkbot.api.config.util.OptionEditor;
+import com.github.manolo8.darkbot.gui.utils.Strings;
+import eu.darkbot.api.config.ConfigSetting;
+
+import javax.swing.*;
+import javax.swing.filechooser.FileNameExtensionFilter;
+import java.awt.*;
+import java.io.File;
+
+public class ImagePicker extends JButton implements OptionEditor<ImageWrapper> {
+    private JFileChooser fc;
+    private ImageWrapper image;
+
+    public ImagePicker(){
+        putClientProperty("JButton.buttonType", "square");
+        addActionListener(e -> {
+            if (fc == null) {
+                String path = image == null ? "." : image.getPath();
+                fc = new JFileChooser(new File(path)) {
+                    @Override
+                    protected JDialog createDialog(Component parent) throws HeadlessException {
+                        JDialog dialog = super.createDialog(parent);
+                        dialog.setAlwaysOnTop(true);
+                        return dialog;
+                    }
+                };
+                FileNameExtensionFilter filter = new FileNameExtensionFilter(
+                        "Images Files", "png", "jpeg", "jpg", "gif");
+                fc.setFileFilter(filter);
+                fc.setAcceptAllFileFilterUsed(false);
+            }
+
+            if (fc.showOpenDialog(null) == JFileChooser.APPROVE_OPTION) {
+                image.setPath(fc.getSelectedFile().getAbsolutePath());
+                setText(Strings.fileName(image.getPath()));
+            }
+        });
+
+    }
+    @Override
+    public void setText(String text) {
+        if(text != null && text.contains("\\") && text.length() > 20){
+            super.setText(".."+text.substring(text.lastIndexOf("\\", text.length()-20)));
```

Comment:
i would ensure the cutting point to always be a file separator, so just search for the file separator char that is closest to a target size (probably a longer on than 20 chars, that looks like a pretty small target, maybe 50 chars?)

Reply:
50 char might be too much, shows the whole directory from c: if the image is in downloads

Reply:
just checked and yeah, 50 as a target may be too much, maybe within the range of 20-30 should be fine